linalg/hypre.hpp

```diff
@@ -52,6 +52,48 @@ namespace mfem
 class ParFiniteElementSpace;
 class HypreParMatrix;
 
+
+/** @brief A simple singleton class for hypre's global settings, that 1) calls
+    HYPRE_Init() and sets some GPU-relevant options at construction and 2) calls
+    HYPRE_Finalize() at destruction. */
+class Hypre
+{
+public:
+   /// @brief Initialize hypre by calling HYPRE_Init() and set default options.
+   /// After calling Hypre::Init(), hypre will be finalized automatically at
+   /// program exit.
+   ///
+   /// Calling HYPRE_Finalize() manually is not compatible with this class.
+   static void Init() { Instance(); }
+
+   /// @brief Finalize hypre (called automatically at program exit if
+   /// Hypre::Init() has been called).
+   ///
+   /// Multiple calls to Hypre::Finalize() have no effect. This function can be
+   /// called manually to more precisely control when hypre is finalized.
+   static void Finalize();
```

Comment:
Why the different doxygen comment styles?

Reply:
I don't have a special preference. Maybe, the more commonly use style in MFEM is like this:
```c++
   /** @brief Multi-line
       brief
       documentation. */
   /** Additional extended documentation. */
   void MyMethod();
```
or similarly:
```c++
   /** @brief Multi-line
       brief
       documentation.

       Additional extended documentation. */
   void MyMethod();
```
@jandrej, is this the style you think we should try to stick with?

Reply:
@pazner, do you have a preference for one of the above commenting styles?

Reply:
My *personal* preference is for `///` comments, but purely as a matter of opinion. For this class it doesn't make a big difference to me.

I would also be fine leaving the two different styles, they will render the same anyway.

Reply:
Addressed in https://github.com/mfem/mfem/pull/2873/commits/d13d86e642c62379eb553dabd2c174718de3666b